### ISSUE TITLE
pknx not launching, must install .NET but it is already installed

### ISSUE THREAD

**Issue by Phyore** [kwsch/pkNX#283]
Trying to run the latest pipeline artifact. Having not updated for a while. It mentioned requiring 6.0.0. Downloaded the runtime on the linked page from the error.

pknx still did not load, same error. Tried a restart, tried uninstalling .NET 6.0.10 and isntalling 6.0.0 and still getting the same error.

OS: Windows 10
Build: https://dev.azure.com/project-pokemon/pkNX/_build/results?buildId=5666&view=results

**Comment by Phyore** [comment]
I think I have solved this. It looks like I needed to install: https://dotnet.microsoft.com/en-us/download/dotnet/6.0 
![image](https://user-images.githubusercontent.com/2222368/197338756-c9bd7ec7-913b-4944-a84d-668bdef5e5eb.png)

And now the app runs. I found this out by looking into the pkNX.runtimeconfig.json

**Comment by JelleInfinity** [comment]
That's the SDK you're installing now though. If that's really needed to run the app I think we need to look at that

**Comment by JelleInfinity** [comment]
![Screenshot_20221022-142850.jpg](https://user-images.githubusercontent.com/60387522/197339045-5f9bcf71-c5c2-4b7a-8b3e-e01bb6f3e7d6.jpg)

I think you only need the desktop runtime

**Comment by Phyore** [comment]
> That's the SDK you're installing now though. If that's really needed to run the app I think we need to look at that

Yeah, installing the first (Linked from the error popup) didn't seem to let me run. Only after installing the SDK did it start working. As I mentioned the runtime config mentioned needing both. (Although I could just be reading that wrong)
![image](https://user-images.githubusercontent.com/2222368/197339112-dbb71154-c48a-46a6-9fd4-13697cebbe9e.png)

**Comment by JelleInfinity** [comment]
The SDK is the source development kit. This includes a whole lot of extra files that you shouldn't need to run the app.

From what I can tell, you installed the .NET runtime, but not the .NET Desktop Runtime.

The second is needed to run windows only components that we use (Win Forms)

**Comment by Phyore** [comment]
> The SDK is the source development kit. This includes a whole lot of extra files that you shouldn't need to run the app.
> 
> From what I can tell, you installed the .NET runtime, but not the .NET Desktop Runtime.
> 
> The second is needed to run windows only components that we use (Win Forms)

You are right! I have uninstalled all those bits and solo installed the desktop framework and it works fine now.

![image](https://user-images.githubusercontent.com/2222368/197339382-42eab3b2-9ca0-42f0-a138-895a123079f2.png)
The error pop up when hitting "yes" takes you to download the .NET runtime and not the .NET Desktop Runtime. I think that is where the problem is then. 

I just assumed it'd take me to the right place ha!

**Comment by JelleInfinity** [comment]
Unfortunately I don't think we can do much about that error. That's part of .NET, not of our code :/

**Comment by Phyore** [comment]
> Unfortunately I don't think we can do much about that error. That's part of .NET, not of our code :/

No worries there! Might be worth throwing a screenshot with an arrow pointing to the right download on the "Download" section. Just to stop anyone else maybe having that issue in future. But glad it is now working! Thanks for the assist!

**Comment by JelleInfinity** [comment]
Yeah I could add that.

No worries, happy to help and happy modding 🙂
